utility/src/main/java/org/oppia/android/util/logging/firebase/FirebaseEventLogger.kt

```diff
@@ -30,8 +31,17 @@ class FirebaseEventLogger(
     firebaseAnalytics.logEvent(eventLog.actionName.toString(), bundle)
     // TODO(#3792): Remove this usage of Locale.
     firebaseAnalytics.setUserProperty(COUNTRY_USER_PROPERTY, Locale.getDefault().displayCountry)
-    firebaseAnalytics.setUserProperty(
-      NETWORK_USER_PROPERTY, connectivityManager.activeNetworkInfo.typeName
-    )
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
```

Comment:
Please add a corresponding test for this case (to differentiate with pre-R behavior).

Reply:
You may need to create custom shadows (you can actually extend the existing shadows, see http://robolectric.org/extending/).

Reply:
@BenHenning  ok, I tried creating a custom one by extending, but the thing is NetworkInfo class itself doesn't have setters for the `typeName`. This is why the developers didn't add that field in Shadow class. 

```kt
@SuppressWarnings("UnusedDeclaration")
@Implements(NetworkInfo::class)
class OppiaShadowNetworkInfo : ShadowNetworkInfo() {

  private lateinit var typeName: String

  companion object {
    fun newInstance(
      type: Int,
      networkState: NetworkInfo.State,
      typeName: String
    ): NetworkInfo {
      val networkInfo = newInstance(
        /* detailedState= */ null,
        /* type= */ type,
        /* subType= */ 0,
        /* isAvailable= */ true,
        /* state= */ networkState
      )
      val shadowNetworkInfo = Shadows.shadowOf(networkInfo) as OppiaShadowNetworkInfo
      shadowNetworkInfo.setTypeName(typeName)
      return networkInfo
    }
  }

  @Implementation
  private fun setTypeName(typeName: String) {
    this.typeName = typeName
  }
}
```

The only way to get the typeName inside the networkInfo is through the constructor. I will look into how the Shadow calls the constructor and if there's a way to pass an argument to it or not.
`Shadow.newInstanceOf(NetworkInfo.class);`

I will update you once I get something on this. All the above mentioned is things I had looked/tried so far.

Reply:
Could we shadow NetworkCapabilities instead and orchestrate it to return a custom NetworkInfo object? It looks like it has a constructor we can call directly.